third_party/nvfuser/csrc/python_frontend/python_bindings.cpp

```diff
@@ -140,18 +140,32 @@ void initNvFuserPythonBindings(PyObject* module) {
           })
       .def(
           "add_output",
-          [](nvfuser::FusionDefinition& self, nvfuser::Scalar output) {
+          [](nvfuser::FusionDefinition& self, nvfuser::Scalar output, c10::optional<nvfuser::Scalar> alias_input = c10::nullopt) {
             FUSER_PERF_SCOPE("FusionDefinition.add_output (scalar)");
-            self.defineRecord(new nvfuser::OutputRecord<Nvf::Val>(
-                {self.recordingState(output())}));
-          })
+            if (alias_input.has_value()) {
+              self.defineRecord(new nvfuser::OutputRecord<Nvf::Val>(
+                  {self.recordingState(output()), self.recordingState(alias_input.value()())}));
+            } else {
+              self.defineRecord(new nvfuser::OutputRecord<Nvf::Val>(
+                  {self.recordingState(output())}));
+            }
+          },
```

Comment:
Might be nice to add a docstring here by inserting a line like
```cpp
"Declare scalar argment 'output' as an output. If given, 'alias_input' will be aliased by 'output', representing in-place modification of 'alias_input'.",
```

Reply:
This could be a place to put a big scary warning.

Reply:
I got rid of the Scalar aliasing.  That did not make sense to me after thinking about it.